### PR TITLE
Buildscript Remake

### PR DESCRIPTION
**Pull request by Desoroxxx**
## 📝 Description

This PR remakes the build script from scratch, converting it to KTS, and updating Gradle.

## 🎯 Goals

- Fix issues like the mod sometime not compiling changes or compiling at all
- Make the build script cleaner
- Prepare for switching to from tags to buildConfig

## ❌ Non Goals

- It is not a goal to make compiling faster

## 🚦 Testing 

Running the client in dev env, building the mod.

## ⏮️ Backwards Compatibility 

This is an internal change that will not impact the mod in a direct manner.

## 📚 Related Issues & Documents

None.

## 🖼️ Screenshots/Recordings

None

## 📖 Added to documentation?

- [ ] 📜 README.md
- [ ] 📑 Documentation
- [ ] 📓 Javadoc
- [x] 🍕 Comments
- [ ] 🙅 No documentation needed

## Summary by CodeRabbit

- **New Features**
	- Introduced configurations for a Minecraft mod project, including dependencies, repositories, run configurations, and task configurations.
- **Chores**
	- Updated Gradle settings for improved task execution and caching.
	- Upgraded Gradle distribution version to enhance build system performance.
	- Set the root project name and configured plugin repositories for better project management.

### CODE REVIEW

**Comment by Desoroxxx** [comment]
@coderabbitai review

**Review by Paneedah — COMMENTED** [review]
Looks good to me

**Review by Paneedah — APPROVED** [review]
For fuck sake work github

File: build.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+            "FMLCorePluginContainsFMLMod" to "true",
+            "ForceLoadAsMod" to "true"
+        )
+    }
+}
+
+tasks.withType<JavaCompile>().configureEach {
+    options.encoding = "UTF-8"
+    options.isFork = true
+    options.forkOptions.jvmArgs = listOf("-Xmx4G")
```
[the remark this answers is not in the document]
Comment (Desoroxxx) [reply build.gradle.kts:173]: No, it was not done in this PR to avoid conflicts

**Review by Paneedah — APPROVED** [review]
//approve I am too lazy to read it

**Pull request merged**